internal/controller/utils/nim.go

```diff
@@ -285,56 +271,80 @@ func requestToken(logger logr.Logger, req *http.Request) (*NimTokenResponse, err
 }
 
 // attemptToPullManifest is used for pulling a runtime for verifying access
-func attemptToPullManifest(logger logr.Logger, runtime NimRuntime, tokenResp *NimTokenResponse) error {
+func attemptToPullManifest(logger logr.Logger, runtime NimRuntime, tokenResp *NimTokenResponse) (*http.Response, error) {
 	req, reqErr := http.NewRequest("GET", fmt.Sprintf(nimGetRuntimeManifestFmt, runtime.Resource, runtime.Version), nil)
 	if reqErr != nil {
-		return reqErr
+		return nil, reqErr
 	}
 
 	req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", tokenResp.Token))
 	req.Header.Add("Accept", "application/vnd.oci.image.index.v1+json")
 
-	resp, respErr := NimHttpClient.Do(req)
-	if respErr != nil {
-		return respErr
-	}
-	logApiResponse(logger, req, resp)
-
-	if resp.StatusCode != http.StatusOK {
-		return fmt.Errorf("failed to pull manifest")
-	}
-
-	return nil
+	return handleRequest(logger, req)
 }
 
 // getModelData is used for fetching NIM model data for the given runtime
 func getModelData(logger logr.Logger, runtime NimRuntime, tokenResp *NimTokenResponse) (*NimModel, string, error) {
 	req, reqErr := http.NewRequest("GET", fmt.Sprintf(nimGetNgcModelDataFmt, runtime.Org, runtime.Team, runtime.Image), nil)
 	if reqErr != nil {
+		logger.Error(reqErr, "failed to construct request")
```

Comment:
The caller of the getModelData in this PR is not checking returned errors, maybe this part can be improved.

Reply:
Yes. The [caller](https://github.com/TomerFi/odh-model-controller/blob/fix-eu-content-unavailable/internal/controller/utils/nim.go#L154) uses the error as a Boolean flag, we can probably move the logger statement to the caller. This might make more sense.

Reply:
We can't move it up to the caller. If we do, we'll end up with repeating log records. The next if statement from where you're looking, [line 297](https://github.com/TomerFi/odh-model-controller/blob/fix-eu-content-unavailable/internal/controller/utils/nim.go#L297), returns an error without logging because the handleRequest function does the logging. So, in this case, if the caller does logging as well, we'll have double records.

So wait a sec, let me see if I can refactor this, we can probably live without this error being propagated back up.

Reply:
That was a very good catch! Resolved here: https://github.com/opendatahub-io/odh-model-controller/pull/390/commits/4c5b04d41c900cd9ca72935d75da1aa84dbcdcf9.

Reply:
yep :)
thanks